From: Yusuke Endoh <mame@ruby-lang.org>
Date: Tue, 23 Mar 2021 12:20:58 +0900
Subject: test/zlib/test_zlib.rb: stop a failure on armv7l

The platform defines a constant File::TMPFILE, but it seems unavailable
(maybe depending on the file system?).

http://rubyci.s3.amazonaws.com/scw-9d6766/ruby-master/log/20210322T171707Z.fail.html.gz

This change adds some rescue cluases to the test, copied from
test/ruby/test_file.rb.
---
 test/zlib/test_zlib.rb | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/test/zlib/test_zlib.rb b/test/zlib/test_zlib.rb
index c469d24..93c3b2d 100644
--- a/test/zlib/test_zlib.rb
+++ b/test/zlib/test_zlib.rb
@@ -806,6 +806,12 @@ if defined? Zlib
           gz = Zlib::GzipReader.new(io)
           assert_raise(NoMethodError) { gz.path }
         end
+      rescue Errno::EINVAL
+        skip 'O_TMPFILE not supported (EINVAL)'
+      rescue Errno::EISDIR
+        skip 'O_TMPFILE not supported (EISDIR)'
+      rescue Errno::EOPNOTSUPP
+        skip 'O_TMPFILE not supported (EOPNOTSUPP)'
       end
     end
   end
